### PR TITLE
Add heading fragment links

### PR DESCRIPTION
**Pull request by zadeviggers**
Adds fragment links to headings using the `rehype-autolink-headings` plugin.

Video example:

https://user-images.githubusercontent.com/74938858/160072462-2487290b-2a15-4f91-af21-126af05d0b20.mp4

### CODE REVIEW

File: public/index.css [hunk 1]
```diff
@@ -342,11 +360,11 @@ h2.heading {
 	padding-left: 3rem;
 }
 
-[dir="rtl"] .header-link.depth-3 {
```
Comment (zadeviggers) [inline public/index.css:345]: Not sure why this changed - probably prettier being derpy

File: public/index.css [hunk 2]
```diff
@@ -46,10 +46,9 @@ nav ul {
 }
 
 .content > section > * + :is(ul, ol) {
-	margin-top: 0.30rem;
```
Comment (zadeviggers) [inline public/index.css:49]: This was probably also prettier

**Review by zadeviggers — COMMENTED** [review]
Making a review so that my comments show up

**Comment by sarah11918** [comment]
I like this! Preview seems to work fine for me, and colour works in both light/dark modes as far as I can tell.

**Comment by zadeviggers** [comment]
Fixed the merge conflict, should be good to go now.

**Comment by FredKSchott** [comment]
LGTM! But it looks like there's a merge conflict. @zadeviggers If you can resolve, @sarah11918 or I can merge!

**Comment by zadeviggers** [comment]
I think the conflicts should be fixed now. I followed the GitHub instructions, but those are some pretty big diffs so I may very well have messed something up.

**Comment by sarah11918** [comment]
@zadeviggers Heads up, on all smaller screen sizes  I *am* seeing the symbols in the "On this page." Seems to be only full desktop mode where the pilcrow is successfully prevented.

Screenshot 2022-03-29 09 11 47
Screenshot 2022-03-29 09 12 55

**Comment by zadeviggers** [comment]
Oh whoops, I know why that's happening.

**Comment by sarah11918** [comment]
@zadeviggers The deploy preview for the latest change is giving me a 404 for the main page. Can you take a look at that?

**Comment by zadeviggers** [comment]
After inspecting the deploy log, it looks like there was an error but the deploy succeeded anyway for some reason.

**Comment by sarah11918** [comment]
Looking great now, thanks!

**Pull request merged**